flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/BufferPersisterImpl.java

```diff
@@ -203,7 +204,11 @@ private void write(Buffer buffer) throws IOException {
 				MemorySegment segment = buffer.getMemorySegment();
 				int numBytes = buffer.getSize();
 
-				currentOutputStream.write(segment.getArray(), offset, numBytes);
+				if (readBuffer.length < numBytes) {
```

Comment:
What is the motivation of this change? 
Can we use the previous way of writing from backend bytes in segment directly to avoid copying to new bytes?

Reply:
It didn't work with offheap segments (will throw unsupported operation exception). There is probably a better solution.